### PR TITLE
[JAVA] Remove deprecated event-related classes

### PR DESCRIPTION
**Pull request by rachel-mack**
# Pull Request Info

[PR Reviewing Guidelines](https://github.com/mongodb/docs-java/blob/master/REVIEWING.md)

JIRA - <https://jira.mongodb.org/browse/DOCSP-34456>
Staging - <https://docs-mongodbcom-staging.corp.mongodb.com/java/docsworker-xlarge/DOCSP-34456/upgrade/#version-5.0-breaking-changes>

## Self-Review Checklist

- [x] Is this free of any warnings or errors in the RST?
- [x] Did you run a spell-check?
- [x] Did you run a grammar-check?
- [x] Are all the links working?
- [x] Are the [facets and meta keywords](https://wiki.corp.mongodb.com/display/DE/Docs+Taxonomy) accurate?

### CODE REVIEW

File: source/upgrade.txt [hunk 1]
```diff
@@ -116,6 +116,20 @@ Version 5.0 Breaking Changes
   .. After the 5.0 Scala API docs are released, this line will be uncommented. 
     For more information, see the `Observable trait documentation <https://mongodb.github.io/mongo-java-driver/5.0/apidocs/mongo-scala-driver/org/mongodb/scala/Observable.html>`__.
 
+- This driver version removes the following classes from the
+  ``com.mongodb.event`` package:
```
Comment (mongoKart) [inline source/upgrade.txt:120]: S: [My PR](https://github.com/mongodb/docs-java/pull/510/files#diff-9c988395fdf1141e52aa9fa946e169ad1d9fb7de62fd44fc1b95bc1885b9506dR64) hasn't been merged yet, but it changes the language used to introduce these bullet points. We can either leave this bullet point as is for now and come back later to reword, or change it as part of this ticket to match the changes in my PR.

**Review by mongoKart — APPROVED** [review]
lgtm + suggestion

File: source/upgrade.txt (hunk 1 above)
Comment (rachel-mack) [reply source/upgrade.txt:120]: Ya, there's going to be a few changes required to several PRs - once yours is merged, we can start rebasing the other ones, and account for those changes.

File: source/upgrade.txt [hunk 2]
```diff
@@ -152,6 +152,20 @@ Version 5.0 Breaking Changes
...
+  
+  - ``ClusterListenerAdapter``
+  - ``ConnectionAddedEvent``
+  - ``ConnectionPoolListenerAdapter``
+  - ``ConnectionPoolOpenedEvent``
+  - ``ConnectionRemovedEvent``
+  - ``ServerListenerAdapter``
+  - ``ServerMonitorListenerAdapter``
```
Comment (stIncMale) [inline source/upgrade.txt:164]: `ConnectionAddedEvent`, `ConnectionPoolOpenedEvent`, `ConnectionRemovedEvent` are events, so the corresponding methods of `ConnectionPoolListener` were also removed: `connectionAdded`, `connectionPoolOpened`, `connectionRemoved` (unfortunately, they are not specified in the description of DOCSP-34456). Let's mention that these methods were removed.

**Review by stIncMale — CHANGES_REQUESTED** [review]

File: source/upgrade.txt [hunk 3]
```diff
@@ -152,6 +152,27 @@ Version 5.0 Breaking Changes
...
+  - ``ConnectionAddedEvent``
+  - ``ConnectionPoolOpenedEvent``
+  - ``ConnectionRemovedEvent``
+  - ``ClusterListenerAdapter``
+  - ``ConnectionPoolListenerAdapter``
+  - ``ServerListenerAdapter``
+  - ``ServerMonitorListenerAdapter``
+
+  Because of these removals, the following methods were also removed from the
+  ``ConnectionPoolListener`` class:
```
Comment (stIncMale) [inline source/upgrade.txt:167]:
```suggestion
  ``ConnectionPoolListener`` interface:
```

**Review by stIncMale — CHANGES_REQUESTED** [review]

**Review by stIncMale — APPROVED** [review]

**Pull request merged**